src/v0/destinations/klaviyo/batchUtil.js

```diff
@@ -0,0 +1,186 @@
+const lodash = require('lodash');
+const { defaultBatchRequestConfig, getSuccessRespEvents } = require('../../util');
+const { JSON_MIME_TYPE } = require('../../util/constant');
+const { BASE_ENDPOINT, CONFIG_CATEGORIES, MAX_BATCH_SIZE, revision } = require('./config');
+const { buildRequest, getSubscriptionPayload } = require('./util');
+/**
+ * This function groups the subscription responses on list id
+ * @param {*} subscribeResponseList
+ * @returns
+ * Example subsribeResponseList =
+ * [
+ * { payload: {id:'list_id', profile: {}}, metadata:{} },
+ * { payload: {id:'list_id', profile: {}}, metadata:{} }
+ * ]
+ */
+const groupSubscribeResponsesUsingListIdV2 = (subscribeResponseList) => {
+  const subscribeEventGroups = lodash.groupBy(
+    subscribeResponseList,
+    (event) => event.payload.listId,
+  );
+  return subscribeEventGroups;
+};
+
+/**
+ * This function returns the list of profileReq which do not metadata common with subcriptionMetadataArray
+ * @param {*} profileReq
+ * @param {*} subscriptionMetadataArray
+ * @returns
+ */
+const getRemainingProfiles = (profileReq, subscriptionMetadataArray) => {
+  const subscriptionListJobIds = subscriptionMetadataArray.map((metadata) => metadata.jobId);
+  return profileReq.filter((profile) => !subscriptionListJobIds.includes(profile.metadata.jobId));
+};
+
+/**
+ * This function builds all the profile requests whose metadata is not there in subscriptionMetadataArray
+ * @param {*} profileRespList
+ * @param {*} subscriptionMetadataArray
+ * @param {*} destination
+ * @returns
+ */
+const getProfiles = (profileRespList, subscriptionMetadataArray, destination) => {
+  const profiles = [];
+  const remainingProfileReq = getRemainingProfiles(profileRespList, subscriptionMetadataArray);
+  remainingProfileReq.forEach((input) => {
+    profiles.push(
+      getSuccessRespEvents(
+        buildRequest(input.payload, destination, CONFIG_CATEGORIES.IDENTIFYV2),
+        [input.metadata],
+        destination,
+      ),
+    );
+  });
+  return profiles;
+};
+
+/**
+ * This function takes susbscriptions as input and batches them into a single request body
+ * @param {events}
+ * events= [
+ * { payload: {id:'list_id', profile: {}}, metadata:{} },
+ * { payload: {id:'list_id', profile: {}}, metadata:{} }
+ * ]
+ */
+const generateBatchedSubscriptionRequest = (events, destination) => {
+  const batchEventResponse = defaultBatchRequestConfig();
+  const metadata = [];
+  // fetching listId from first event as listId is same for all the events
+  const listId = events[0].payload?.listId;
+  const profiles = []; // list of profiles to be subscribes
+  // Batch profiles into dest batch structure
+  events.forEach((ev) => {
+    profiles.push(...ev.payload.profile);
```

Comment:
`ev.payload.profile` is an array or object?

Reply:
it is an object

Reply:
then this syntax is wrong, can you write an test case for this

Reply:
ok got confused with another profiles object